.github/workflows/build.yml

```
@@ -0,0 +1,36 @@
+          cache: 'pip'
+      - name: Install test dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install coverage pytest
+      - name: Run tests with coverage
+        run: |
+          coverage run --source=wifi_manager --omit=wifi_manager/setup.py -m pytest -q --junitxml=pytest.xml
+          coverage xml --omit=wifi_manager/setup.py
+          coverage report -m --omit=wifi_manager/setup.py
+      - name: SonarQube Scan
+        if: ${{ secrets.SONAR_TOKEN != '' }}
```

actionlint (1.7.12)

[error] context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)